drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/DMNScenarioValidation.java

```diff
@@ -0,0 +1,126 @@
+import org.kie.dmn.core.impl.BaseDMNTypeImpl;
+import org.kie.dmn.feel.lang.Type;
+
+import static org.drools.scenariosimulation.backend.util.DMNSimulationUtils.extractDMNModel;
+import static org.drools.scenariosimulation.backend.util.DMNSimulationUtils.extractDMNRuntime;
+import static org.drools.workbench.screens.scenariosimulation.backend.server.util.DMNUtils.getRootType;
+import static org.drools.workbench.screens.scenariosimulation.backend.server.util.DMNUtils.navigateDMNType;
+import static org.drools.workbench.screens.scenariosimulation.model.FactMappingValidationError.createFieldChangedError;
+import static org.drools.workbench.screens.scenariosimulation.model.FactMappingValidationError.createGenericError;
+import static org.drools.workbench.screens.scenariosimulation.model.FactMappingValidationError.createNodeChangedError;
+import static org.kie.dmn.feel.lang.types.BuiltInType.CONTEXT;
+import static org.kie.dmn.feel.lang.types.BuiltInType.UNKNOWN;
+
+public class DMNScenarioValidation extends AbstractScenarioValidation {
+
+    public static DMNScenarioValidation INSTANCE = new DMNScenarioValidation();
+
+    /**
+     * Validate structure of a DMN test scenario.
+     * Supported checks for each column:
+     * - empty column skip
+     * - DMN node removed
+     * - simple type becomes complex type
+     * - navigation of data type still valid
+     * - field type changed
+     * @param simulation
+     * @param kieContainer
+     * @return
+     */
+    @Override
+    public List<FactMappingValidationError> validate(Simulation simulation, KieContainer kieContainer) {
+        List<FactMappingValidationError> errors = new ArrayList<>();
+        String dmnFilePath = simulation.getSimulationDescriptor().getDmnFilePath();
+        DMNModel dmnModel = getDMNModel(kieContainer, dmnFilePath);
+
+        for (FactMapping factMapping : simulation.getSimulationDescriptor().getFactMappings()) {
+            if (isToSkip(factMapping)) {
+                continue;
+            }
+
+            String nodeName = factMapping.getFactIdentifier().getName();
+
+            DMNType rootDMNType;
+            try {
+                rootDMNType = dmnModel.getDecisionByName(nodeName) != null ?
+                        dmnModel.getDecisionByName(nodeName).getResultType() :
+                        dmnModel.getInputByName(nodeName).getType();
+            } catch (NullPointerException e) {
+                errors.add(createNodeChangedError(factMapping, "node not found"));
+                continue;
+            }
+
+            List<String> steps = expressionElementToString(factMapping);
+
+            // error if direct mapping (= simple type) but it is a composite
+            // NOTE: context is a special case so it is composite even if no fields are declared
+            Type rootType = getRootType((BaseDMNTypeImpl) rootDMNType);
+            if (!CONTEXT.equals(rootType) && steps.size() == 0 && rootDMNType.isComposite()) {
```

Comment:
We could use `steps.isEmpty()` as we do in `RULEScenarioValidation`

Reply:
Done